mongomock/filtering.py

```diff
@@ -408,12 +413,27 @@ def func(doc_val, search_val):
     return func
 
 
-def _type_op(doc_val, search_val):
+def _type_op(doc_val, search_val, in_array=False):
     if search_val not in TYPE_MAP:
         raise OperationFailure('%r is not a valid $type' % search_val)
     elif TYPE_MAP[search_val] is None:
         raise NotImplementedError('%s is a valid $type but not implemented' % search_val)
-    return isinstance(doc_val, TYPE_MAP[search_val])
+
+    spec = TYPE_MAP[search_val]
+    if not isinstance(spec, tuple):
+        spec = (spec,)
+
+    for t in spec:
+        if callable(t):
+            if isinstance(t, type):
+                if isinstance(doc_val, t):
+                    return True
+            elif t(doc_val):
+                return True
+
+    if isinstance(doc_val, list) and not in_array:
```

Comment:
please also check tuples: `if isinstance(doc_val, (list, tuple)) and not in_array:`

Reply:
Done. Thanks!